solutions/usaco-815.mdx

```diff
@@ -0,0 +1,83 @@
+---
+id: usaco-815
+source: USACO Gold 2018 Febuary
+title: Taming the Herd
+author: Shijie Ren
+---
+
+[Official Analysis (C++)](http://www.usaco.org/current/data/sol_taming_gold_feb18.html)
+
+## Explanation
+
+Let $\texttt{dp}[i][j]$ be the minimum answer of first $j$ days with $i$ breakouts.
+
+$j$-th day , if log is tampered and cows would NOT breakout:
+
+$$
+\texttt{dp}[i][j]=\texttt{dp}[i][j-1]+1
+$$
+
+$j$-th day , if log is tampered and cows would breakout :
+
+(So $a[j] \not = 0$)
+
+$$
+\texttt{dp}[i][j]=\texttt{dp}[i-1][j-1]+1
+$$
+
+$j$-th day , if log is NOT tampered:
```

Comment:
On the $j$-th day, there's 3 cases:
1. if [asdf], $relation here$
2. if [asdf]...